event-sourcing/event-store-api/src/main/scala/org/apache/james/eventsourcing/eventstore/History.scala

```diff
@@ -0,0 +1,50 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+package org.apache.james.eventsourcing.eventstore
+
+import org.apache.james.eventsourcing.{Event, EventId}
+
+import scala.annotation.varargs
+import scala.jdk.CollectionConverters._
+
+object History {
+  def empty: History = new History(Nil)
+
+  def of(events: List[Event]): History = new History(events)
+  def of(events: java.util.List[Event]): History = new History(events.asScala.toList)
+
+  @varargs
+  def of(events: Event*): History = of(events.toList)
+}
+
+case class History private(events: List[Event]) {
+  if (hasEventIdDuplicates(events)) throw EventStoreFailedException("Event History contains duplicated EventId")
```

Comment:
i would vote for yes. Because `if` in scala is an expression and not a statement like in java.
IHMO it make sense to be able to express simple if on one line. 
(this particular example is not the best example)

Reply:
for me it breaks readibility: we don't know at first where does finish the condition and when starts the expression (this is an argument for a line break). And then when you have a line break you need brackets because if you add an expression you will not notice the missing brackets and you will break your condition.

Reply:
I added the bracket in this case. I still think that when we do not do side effect it can be more readeable to not have brackets, but i imagine it's a matter of taste which will need to be decided with the team.